### PR TITLE
gather operator

### PR DESCRIPTION
**Pull request by richagadgil**

### CODE REVIEW

File: py/torch_migraphx/fx/converters/acc_ops_converters.py [hunk 1]
```diff
@@ -1135,6 +1135,31 @@ def acc_ops_embedding(mgx_module, node, args, kwargs):
...
 
+@migraphx_converter(acc_ops.gather)
+def acc_ops_gather(mgx_module, node, args, kwargs):
+    inp = kwargs['input']
+    dim = kwargs['dim']
+    index = kwargs['index']
+
+    assert not inp.is_quantized() and not index.is_quantized()
+    
+    dimensions = index.shape().lens()
```
Comment (shivadbhavsar) [inline py/torch_migraphx/fx/converters/acc_ops_converters.py:1146]:
```suggestion
    index_lens = index.shape().lens()
```
A little nitpick but its a little confusing to keep track of too many similar variables "dimension", "dim", "dims"

File: py/torch_migraphx/fx/converters/acc_ops_converters.py [hunk 2]
```diff
@@ -1135,6 +1135,31 @@ def acc_ops_embedding(mgx_module, node, args, kwargs):
...
+    if dim < 0:
+        dim = len(dimensions) + dim
+    
+    dims = [torch.arange(0, i) for i in dimensions]
+
+    import itertools
+    tensor = torch.tensor(list(itertools.product(*dims)))
```
Comment (shivadbhavsar) [inline py/torch_migraphx/fx/converters/acc_ops_converters.py:1153]: Call `tensor` something a bit more meaningful here? maybe `base_coords`?

File: py/torch_migraphx/fx/converters/aten_ops_converters.py [hunk 3]
```diff
@@ -1295,3 +1295,10 @@ def aten_ops_erf(mgx_module, node, args, kwargs):
     acc_kwargs = {"input": args[0]}
     return acc_ops_converters.acc_ops_erf(mgx_module, node, (), acc_kwargs)
 
+
+@migraphx_converter(torch.ops.aten.gather.default)
+def aten_ops_erf(mgx_module, node, args, kwargs):
```
Comment (shivadbhavsar) [inline py/torch_migraphx/fx/converters/aten_ops_converters.py:1300]:
```suggestion
def aten_ops_gather(mgx_module, node, args, kwargs):
```

File: tests/dynamo/converters/test_gather_dynamo.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+                                            ((2, 3, 4, 5, 6), -4)])
+def test_gather(input_dim, dim):
+    input = torch.rand(input_dim).cuda()
+
+    dim_size = input.size(dim)
+    index_shape = list(input.size())
+    index_shape[dim] = np.random.randint(1, dim_size)  
+    index = torch.randint(0, dim_size, index_shape).cuda()
+
+    mod = FuncModule(torch.gather, dim=dim, index=index).cuda()
```
Comment (shivadbhavsar) [inline tests/dynamo/converters/test_gather_dynamo.py:21]: gotta call the aten function here

File: test_gather_final.py [hunk 5]
```diff
@@ -0,0 +1,112 @@
+import numpy as np
```
Comment (shivadbhavsar) [inline test_gather_final.py:1]: remove file from PR. This is very nice though, we should find a place for it

**Review by shivadbhavsar — COMMENTED** [review]
Looks good! Just some small things

**Review by shivadbhavsar — APPROVED** [review]
